### PR TITLE
Create controller to auto register TPRs with the aggregator

### PR DESCRIPTION
**Pull request by deads2k**
Builds on https://github.com/kubernetes/kubernetes/pull/42732 (already lgtmed)

Creates a simple controller to wire TPRs with the API Service autoregistration controller.

@kubernetes/sig-api-machinery-misc @ncdc

### CODE REVIEW

File: pkg/master/thirdparty/tprregister_controller.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+func (c *tprRegisterController) enqueueTPR(tpr *extensions.ThirdPartyResource) {
+	_, group, err := thirdpartyresourcedata.ExtractApiGroupAndKind(tpr)
+	if err != nil {
+		utilruntime.HandleError(err)
+		return
+	}
+	for _, version := range tpr.Versions {
+		c.queue.Add(schema.GroupVersion{Group: group, Version: version.Name})
+	}
+}
```
Comment (ncdc) [inline pkg/master/thirdparty/tprregister_controller.go]: add a newline

File: pkg/master/thirdparty/tprregister_controller.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+			if version.Name == groupVersion.Version && group == groupVersion.Group {
+				hasResources = true
+				break
+			}
+		}
+	}
+
+	apiServiceName := groupVersion.Version + "." + groupVersion.Group
+
+	if !hasResources {
```
Comment (ncdc) [inline pkg/master/thirdparty/tprregister_controller.go]: What about if the cache is slow to sync? You could have a new TPR enqueued but it's not in the lister yet?

Comment (deads2k) [inline pkg/master/thirdparty/tprregister_controller.go]: > What about if the cache is slow to sync? You could have a new TPR enqueued but it's not in the lister yet?

The cache is always at least as current as the enqueue for the same type.

**Comment by deads2k** [comment]
lgtm via irc.  vendor/BUILD strikes again, so approving

**Comment by ncdc** [comment]
Recording my lgtm 😄

**Pull request merged**